common/lib/xmodule/xmodule/html_module.py

```diff
@@ -417,6 +418,35 @@ class CourseInfoModule(CourseInfoFields, HtmlModuleMixin):
     # statuses
     STATUS_VISIBLE = 'visible'
     STATUS_DELETED = 'deleted'
+    TEMPLATE_DIR = 'courseware'
+
+    @XBlock.supports("multi_device")
+    def student_view(self, _context):
+        """
+        Return a fragment that contains the html for the student view
+        """
+        return Fragment(self.get_html())
+
+    def get_html(self):
+        """
+        When we switch this to an XBlock, we can merge this with student_view,
+        but for now the XModule mixin requires that this method be defined.
+        """
+        # pylint: disable=no-member
+        if self.data != "":
+            if self.system.anonymous_student_id:
+                return self.data.replace("%%USER_ID%%", self.system.anonymous_student_id)
+            return self.data
+        else:
+            course_updates = [item for item in self.items if item.get('status') == self.STATUS_VISIBLE]
```

Comment:
Generators are usually a bit faster than building a list in this sort of situation where you're just constructing an intermediate result.